llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -4018,6 +4018,16 @@ foldICmpOfCmpIntrinsicWithConstant(ICmpInst::Predicate Pred, IntrinsicInst *I,
       NewPredicate = ICmpInst::ICMP_ULE;
     break;
 
+  case ICmpInst::ICMP_ULT:
+    if (C.ugt(1) && !C.isAllOnes())
```

Comment:
```suggestion
    if (C.ugt(1))
```
This check doesn't appear to be necessary: https://alive2.llvm.org/ce/z/NyMjpe

Reply:
Yep, fixed.